cloud-controller-manager/do/resources.go

```diff
@@ -43,8 +43,10 @@ type tagMissingError struct {
 }
 
 type resources struct {
-	clusterID    string
-	clusterVPCID string
+	clusterID                string
+	clusterVPCID             string
+	publicAccessFirewallName string
+	publicAccessFirewallTags []string
```

Comment:
Let's define a new type `publicAccessFirewall` with fields for the firewall name and the tags, and make it a member of `resources`:

```go
type publicAccessFirewall struct {
    name string
    tags []string
}

type resources struct {
    [...]
    firewall publicAccessFirewall
}
```

This nicely bundles the firewall-specific parameters, allows for better extensibility, and minimizes the noise on the various `newResources()` invocations we needed to to update in this PR.

Reply:
This is a great suggestion!